### ISSUE TITLE
missed reporting multiple occurance issue recursively on a file 

### ISSUE THREAD

**Issue by prakashmirji** [golangci/golangci-lint#2176]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

After running the tool on our go project, it only detected the first occurance of the linting issue in a file. It missed to reporting second occurance of the same issue in the same file. I was hoping tool will report all occurrences. Is it expected behavior?  See more details below.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# paste output here
```
golangci-lint has version 1.41.1 built from a2074809 on 2021-06-19T16:01:50Z
</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
```
don't have this file
</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```
go version go1.16.6 darwin/amd64
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/prakash/Library/Caches/go-build"
GOENV="/Users/prakash/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/prakash/go-workspace/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/prakash/go-workspace"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GOVCS=""
GOVERSION="go1.16.6"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/prakash/go-workspace/src/github.com/abc/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/zb/_bc7h9cj0nxf9cf8jc8k_kgw0000gn/T/go-build2607357916=/tmp/go-build -gno-record-gcc-switches -fno-common"
</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```
I am referring to this issue

pkg/config/k8sconfigs.go:65:2: S1021: should merge variable declaration with assignment on next line (gosimple)
        var mapHosts map[string]string

The same file had another instance of this, but it was not part of the output. Looks like only first occurrence is detected.

</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```

        var mapHosts map[string]string
	mapHosts = make(map[string]string)
	for _, value := range workerHosts {
		mapHosts[value.InternalName] = value.InternalIP
	}
</details>

**Comment by prakashmirji** [comment]
update: I missed seeing the argument `--max-same-issues` to run command. With this argument, I can now see other occurrences. My bad. Can I close this issue now?
